build_projects/dotnet-host-build/PublishTargets.cs

```diff
@@ -83,21 +85,81 @@ public static BuildTargetResult FinalizeBuild(BuildTargetContext c)
             {
                 string targetContainer = $"{Channel}/Binaries/Latest/";
                 string targetVersionFile = $"{targetContainer}{CommitHash}";
+                string flagFinalizationInProgress = $"{targetContainer}/Finalizing";
                 string semaphoreBlob = $"{Channel}/Binaries/sharedFxPublishSemaphore";
+                
                 AzurePublisherTool.CreateBlobIfNotExists(semaphoreBlob);
 
-                string leaseId = AzurePublisherTool.AcquireLeaseOnBlob(semaphoreBlob);
```

Comment:
That loop magic is to keeping trying to acquire the lease on the blob. Here, I am trying to not even get into that loop if we have finalization going on for a different build - effectively at two different levels and if this works, we may want to remove that loop completely from lease acquisition.